src/agoradatatools/etl/transform/immunohisto_transform.py

```diff
@@ -65,6 +66,231 @@ def prepare_immunohisto_data(df: pd.DataFrame) -> pd.DataFrame:
     return df
 
 
+def round_y_axis_max(y_axis_max: Union[int, float, str]) -> float:
+    """
+    This function rounds the y_axis_max value to the nearest sensible nice round number.
+
+    Logic:
+    - If max == 0, then y_axis_max == 10
+    - Else, round UP to the next "nice" number where the second digit is 0 or 5
+
+    Examples:
+    - 0.0021 rounds up to 0.0025
+    - 0.0004 rounds up to 0.00045
+    - 0.329486078 rounds up to 0.35
+    - 0.089 rounds up to 0.090
+    - 1094 rounds up to 1500
+    - 1322498 rounds up to 1500000
+    - 728591 rounds up to 750000
+    - 3973 rounds up to 4000
+    - 1.616 rounds up to 2.0
+    """
+    # Convert to float if it's a string or other type
+    try:
+        y_axis_max = float(y_axis_max)
+    except (ValueError, TypeError):
+        # If conversion fails, return 10 (default case)
+        return 10.0
+
+    # Handle special cases: zero or negative values
+    if y_axis_max <= 0:
+        return 10.0 if y_axis_max == 0 else 0.0
+
+    # Find the order of magnitude of the number
+    magnitude = int(math.floor(math.log10(y_axis_max)))
+
+    # Scale the number so the first digit is in the ones place
+    scaled = y_axis_max / (10**magnitude)
+
+    # Extract first digit (leftmost) and second digit
+    first_digit = int(scaled)
+
+    # Use string method to avoid floating point precision issues
+    # Use a robust approach that adapts to the number's precision needs
+    # For very small numbers, we need more decimal places to capture the second digit accurately
+    # Use a precision that's sufficient for the expected range of values
+    precision = max(15, int(-math.log10(abs(scaled)) + 5)) if scaled != 0 else 15
+    scaled_str = f"{scaled:.{precision}f}"
+
+    if "." in scaled_str:
+        decimal_part = scaled_str.split(".")[1]
+        if len(decimal_part) >= 1:
+            second_digit = int(decimal_part[0])
+        else:
+            second_digit = 0
+    else:
+        second_digit = 0
+
+    # Always round UP to the next "nice" number
+    # Nice numbers have second digit of 0 or 5
+    if second_digit == 0:
+        # Already a nice number, but we need to round UP
+        # So we go to the next nice number
+        rounded_second = 5
+    elif second_digit <= 5:
+        # Round up to 5
+        rounded_second = 5
+    else:
+        # Round up to next first digit with 0
+        rounded_second = 0
+        first_digit += 1
+
+    # Handle edge case where first digit rounded up to 10
+    if first_digit >= 10:
+        first_digit = 1
+        magnitude += 1
+
+    # Construct the result
+    result = (first_digit + rounded_second / 10.0) * (10**magnitude)
+
+    return result
+
+
+def _calculate_y_axis_max_map(
+    dataset: pd.DataFrame,
+) -> Dict[Tuple[str, str, str], float]:
+    """
+    Calculate y_axis_max for each combination of (name, evidence_type, tissue) across all ages.
+
+    Args:
+        dataset: The prepared dataset
```

Comment:
```suggestion
        dataset: The prepared dataset. It must have columns "name", "evidence_type", "tissue", and "value".
```
Since this data frame has already been validated to have these columns in the main transform function, I don't think we need to add error checking here, but I think it's helpful if the comments explicitly say what columns are used here.